src/classes/PromoModals.js

```diff
@@ -33,23 +35,51 @@ class PromoModals {
 
 	static isTimeForAPlusPromo() { return this._isTimeForAPromo(PLUS); }
 
+	static isTimeForInsightsPromo() { return this._isTimeForAPromo(INSIGHTS); }
+
 	static recordPlusPromoSighting() { this._recordPromoSighting(PLUS); }
 
-	// TODO integrate the Insights promo modal into the "has it been long enough since last modal?" logic here
+	static recordInsightsPromoSighting() { this._recordPromoSighting(INSIGHTS); }
+
 	static _isTimeForAPromo(type) {
-		const lastSeenTime = conf[`${type}_${PROMO_MODAL_LAST_SEEN}`];
+		const lastSeenPlusPromo = conf[`${PLUS}_${PROMO_MODAL_LAST_SEEN}`];
+		const lastSeenInsightsPromo = conf[`${INSIGHTS}_${PROMO_MODAL_LAST_SEEN}`];
+		const lastSeenPromo = Math.max(lastSeenPlusPromo, lastSeenInsightsPromo);
 
-		if (lastSeenTime === null) { return true; }
+		if (conf.notify_promotions === false) { return false; }
+		if (lastSeenPromo === null) { return true; }
```

Comment:
If `lastSeenPlusPromo` and `ladstSeenInsightsPromo` are both `null`, `Math.max(lastSeenInsightsPromo, lastSeenInsightsPromo)` will return....wait for it.....0 (wtflol). Since `0 !== null`, this will not behave as expected if neither promo has been shown.

(Also, make sure you initialize `insights_promo_modal_last_seen` to `null` and not 0 in `ConfData`).

Reply:
- Moving the early return to the top!
- Javascript is weird! Let's change that line to
![image](https://user-images.githubusercontent.com/10860129/67303632-b4755f80-f4c0-11e9-8d99-021ddf98e8bc.png)
- Initialized insights_promo_modal_last_seen` to null!